lib/Util/Builtin/GccCompilerSupport.in.cpp

```diff
@@ -49,7 +57,10 @@ std::string GccCompilerSupport::compileCppModule(const Pothos::Util::CompilerArg
     }
 
     //add compiler flags
-    args.push_back("-std=c++11");
+    args.push_back(CPP_STD_FLAG);
+#ifndef __GLIBCXX__
+    args.push_back("-stdlib=libc++");
```

Comment:
Will this break GCC variants?

Reply:
Maybe. Havent seen it yet. For your clang build, does cmake automatically come up with "-stdlib=libc++"?

Reply:
The opposite. Ubuntu Clang installations use libstdc++, so specifying libc++ broke things. I'd rather have a runtime check; It's an unfortunate assumption that the same would apply for clients, but I haven't settled on a workaround.

Reply:
So there is probably a way to get the linker flags into a string just like the way the target name is extracted
```
example:
#get the library name into Paths.cpp for getPothosRuntimeLibraryPath()
target_compile_definitions(Pothos PRIVATE "-DPOTHOS_LIBRARY_NAME=\"$<TARGET_FILE_NAME:Pothos>\"")
```

* https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html

And there is a `$<TARGET_PROPERTY:tgt,prop>` and properties like COMPILE_FLAGS and LINK_FLAGS

And just check at runtime if the string with the flags in it contains -stdlib and use it. That will fool proof it to just make it match whatever the cmake project was thinking for the main library.

Reply:
> And just check at runtime if the string with the flags in it contains -stdlib and use it.

well that didnt work like I had hoped

So in the spirit of merging this, did this `ifndef __GLIBCXX__` code actually mess anything up that you know about?

Reply:
I just tested what's currently up, and GCC and Clang both work fine on Linux with this change. I don't have a convenient BSD setup, but in theory it should add the flag as it did before.